src/ima_emulator.rs

```diff
@@ -82,14 +83,14 @@ fn ml_extend(
             }
             Some(search_hash) => {
                 let mut hasher = openssl::sha::Sha1::new();
-                hasher.update(START_HASH);
+                hasher.update(&running_hash);
                 hasher.update(&template_hash);
-                let running_hash: Vec<u8> = hasher.finish().into();
-                let running_hash = Digest::try_from(running_hash)?;
+                running_hash = hasher.finish().into();
+                let digest = Digest::try_from(running_hash.clone())?;
```

Comment:
Using `.as_slice` instead of `.clone` might avoid making an extra copy?

Reply:
Yes, I agree. I'll fix this.